### PR TITLE
address feature

### PR DESCRIPTION
**Pull request by taha7**

### CODE REVIEW

File: src/services/domain/address.service.ts
Comment (usef) [inline src/services/domain/address.service.ts:1]: This is like a model?

File: src/services/domain/address.service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    where: {
+      customerId
+    },
+    data: {
+      default: false
+    }
+  })
+  return prisma.address.create({
+    data: {
+      state: data.state,
```
Comment (usef) [inline src/services/domain/address.service.ts:19]: It isn't a foreign key like the city?

File: src/validations/address.validations.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { object, string } from 'yup'
+
+export const createAddressValidator = object().shape({
+  country: string().required(),
+  address: string().required(),
+  city: string().required(),
```
Comment (usef) [inline src/validations/address.validations.ts:6]: It is an id the services/domain/address.service.ts , why is it a string here?

File: src/services/domain/address.service.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const customerId = 1
+  await prisma.address.updateMany({
+    where: {
+      customerId
+    },
+    data: {
+      default: false
+    }
+  })
+  return prisma.address.create({
```
Comment (usef) [inline src/services/domain/address.service.ts:17]: Wouldn't this throw an error if the ids are duplicate? Or in another case it'd update and create in the same function.

File: src/services/domain/address.service.ts
Comment (taha7) [reply src/services/domain/address.service.ts:1]: No, I meant a service related to our business domain

File: src/services/domain/address.service.ts (hunk 3 above)
Comment (taha7) [reply src/services/domain/address.service.ts:17]: Didn't get what you meant, but let me try to explain

One customer can have many addresses, so we would need one of them to be the default one.
For every new address creation, we would need to make all addresses default value "false" and the new one will be the default

File: src/validations/address.validations.ts (hunk 2 above)
Comment (taha7) [reply src/validations/address.validations.ts:6]: We receive it from the frontend as string

**Review by usef — APPROVED** [review]

**Review by mosafa697 — APPROVED** [review]

**Review by mosafa697 — APPROVED** [review]

**Pull request merged**